### PR TITLE
feat(store): Throw if store not initialized

### PR DESCRIPTION
**Pull request by rozzilla**

### CODE REVIEW

File: asyncforge.js [hunk 1]
```diff
@@ -36,7 +36,10 @@ function memo (name) {
 
   function get () {
     const store = asyncLocalStorage.getStore()
-    return store?.[sym]
+    if (!store) {
+      throw new Error('asyncforge store is not initialized')
```
Comment (mcollina) [inline asyncforge.js:40]: Can you add the name of the memo here if not anonymous.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: test/do-not-start-memo.test.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ test('nested', async (t) => {
   const p = tspl(t, { plan: 5 })
   const a = memo()
 
-  p.deepStrictEqual(a(), undefined)
+  p.throws(() => a(), { message: 'asyncforge store is not initialized for memo1' })
```
Comment (mcollina) [inline test/do-not-start-memo.test.js:45]: please add a test for named memo

Comment (rozzilla) [reply test/do-not-start-memo.test.js:54]: [9091370](https://github.com/mcollina/asyncforge/pull/3/commits/90913709e5563c0ac46fb7232b9cf5f4a6bbc7f6)

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**